UnityProject/Assets/Scripts/Objects/Other/PineTreeXmas.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AddressableReferences;
+using Audio.Containers;
+using Grpc.Core;
+using Managers;
+using Mirror;
+using ScriptableObjects.TimedGameEvents;
+using UnityEngine;
+
+namespace Objects.Other
+{
+	public class PineTreeXmas : NetworkBehaviour, ICheckedInteractable<HandApply>
+	{
+		[SerializeField] private SpriteDataSO xmasSpriteSO;
+		[SerializeField] private TimedGameEventSO eventData;
+		[SerializeField] private GameObject giftObject;
+		[SerializeField] private AddressableAudioSource ambientReminder;
+
+		[SyncVar] private List<string> giftedPlayers = new List<string>();
+		private bool canPickUpGifts;
+
+		private SpriteHandler spriteHandler;
+
+		private void Awake()
+		{
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			if (TimedEventsManager.Instance.ActiveEvents.Contains(eventData))
+			{
+				canPickUpGifts = true;
+				spriteHandler.SetSpriteSO(xmasSpriteSO);
+				PlayEasterEgg();
+			}
+		}
+
+		private async void PlayEasterEgg()
+		{
+			await Task.Delay(60 * 1000); // millaseconds
+			if (DMMath.Prob(10))
+			{
+				_ = SoundManager.PlayNetworkedAtPosAsync(ambientReminder, gameObject.WorldPosServer());
+			}
+			PlayEasterEgg();
+		}
```

Comment:
i'm not sure i like this, does the task get removed when the object is destroyed? Might need to add a null check for `this` and `gameObject`

Reply:
Might be better to just do a 60 second periodic update